doc/cascadia/Command-Palette-spec.md

```diff
@@ -0,0 +1,277 @@
+        { "icon": null, "name": "Close Pane", "action": "closePane" },
+        { "icon": null, "name": "[-] Split Horizontal", "action": "splitHorizontal" },
+        { "icon": null, "name": "[ | ] Split Vertical", "action": "splitVertical" },
+        { "icon": null, "name": "Next Tab", "action": "nextTab" },
+        { "icon": null, "name": "Prev Tab", "action": "prevTab" },
+        { "icon": null, "name": "Open Settings", "action": "openSettings" },
+        { "icon": null, "name": "Open Media Controls", "action": "openTestPane" }
+    ],
+```
+
+* "open" should return both "**Open** Settings" and "**Open** Media Controls".
+* "Tab" would return "New **Tab**", "Close **Tab**", "Next **Tab**" and "Prev
+  **Tab**".
+* "P" would return "Close **P**ane", "[-] S**p**lit Horizontal", "[ | ]
+  S**p**lit Vertical", "**P**rev Tab", "O**p**en Settings" and "O**p**en Media
+  Controls".
+* Even more powerfully, "sv" would return "[ | ] **S**plit **V**ertical". This
+  is a great example of how a user could execute a command with very few
+  keystrokes.
+
+As the user types, we should **bold** each matching character in the command
+name, to show how their input correlates to the results on screen.
+
+## Capabilities
+
+### Accessibility
+
+As the entire command palette will be a native XAML element, it'll automatically
+be hooked up to the UIA tree, allowing for screen readers to naturally find it.
+ * When the palette is opened, it will automatically recieve focus.
+ * The terminal panes will not be able to be interacted with while the palette
+   is open, which will help keep the UIA tree simple while the palette is open.
+
+### Security
+
+This should not introduce any _new_ security concerns. We're relying on the
+security of jsoncpp for parsing json. Adding new keys to the settings file
+will rely on jsoncpp's ability to securely parse those json values.
+
+### Reliability
+
+We'll need to make sure that invalid commands are ignored. A command could be
+invalid because:
+* it has a null `name`, or a name with the empty string for a value.
+* it has a null `action`, or an action specified that's not an actual
+  `ShortcutAction`.
+
+We'll ignore invalid commands from the user's settings, instead of hard
+crashing. I don't believe this is a scenario that warrants an error dialog to
+indicate to the user that there's a problem with the json.
+
+### Compatibility
+
+We will need to define default _commands_ for all the existing keybinding
+commands. With #754, we could add all the actions (that make sense) as commands
+to the commands list, so that everyone wouldn't need to define them manually.
+
+### Performance, Power, and Efficiency
+
+We'll be adding a few extra XAML elements to our tree which will certainly
+increase our runtime memory footprint while the palette is open.
+
+We'll aditionally be introducing a few extra json values to parse, so that could
+increase our load times (though this will likely be negligible).
+

```

Comment:
One thought I had was how this works for non-English languages, especially those requiring an IME. Does the command palette control use a standard XAML text box/block? Also ensure that the algorithms used have provisions for understanding and testing the input of CJK text.

(Also, maybe we should add # Globalization to the spec template...)